src/coreclr/src/jit/gentree.h

```diff
@@ -7070,14 +7070,15 @@ inline GenTree* GenTree::gtGetOp2IfPresent() const
     return op2;
 }
 
-inline GenTree* GenTree::gtEffectiveVal(bool commaOnly)
+inline GenTree* GenTree::gtEffectiveVal(bool commaOnly /* = false */)
 {
     GenTree* effectiveVal = this;
     for (;;)
     {
+        assert(!effectiveVal->OperIs(GT_PUTARG_TYPE));
```

Comment:
I'm curious why you added this here - that is, I can see why we wouldn't expect to encounter these under a `COMMA`, but I'm not sure why this special assert is needed.

Reply:
We have several similar functions like `gtEffectiveVal`, `gtRetExprVal` and `gtSkipPutArgType`. When we call each of them we often don't expect nodes that can be skipped by two others to be on top.
For example, when we call `gtEffectiveVal` we don't expect something like `GT_RET_EXPR(GT_COMMA)` or `GT_PUTARG_TYPE(GT_COMMA)`, similar for `gtRetExprVal` it should not see `GT_PUTARG_TYPE(GT_PUTARG_TYPE)`.
If we do it means the caller did not check for these nodes and we don't skip what we hoped to skip, it doesn't cause correctness issues in my understanding but introduces asm regressions when we can't parse something like we did before the change.